wire-ios-data-model/Source/MLS/Actions/UpdateConversationProtocolAction.swift

```diff
@@ -0,0 +1,82 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public struct UpdateConversationProtocolAction: EntityAction {
+
+    // MARK: - Properties
+
+    public var qualifiedID: QualifiedID
+    public var messageProtocol: MessageProtocol
+    public var resultHandler: ResultHandler?
+
+    // MARK: - Life cycle
+
+    public init(
+        qualifiedID: QualifiedID,
+        messageProtocol: MessageProtocol
+    ) {
+        self.qualifiedID = qualifiedID
+        self.messageProtocol = messageProtocol
+    }
+
+}
+
+extension UpdateConversationProtocolAction {
```

Comment:
Is there a good reason for extending the struct rather than doing all the implementation in the initial declaration?

Reply:
I like to structure it a bit using extensions, but if you prefer not to do it, I can merge the parts.

One minor advantage might be that all the stored properties are visible at one sight, within a few lines of code.  Extensions cannot declare stored properties.